corehq/apps/change_feed/consumer/feed.py

```diff
@@ -163,10 +165,19 @@ def _filter_offsets(self, offsets) -> Optional[Dict[TopicPartition, int]]:
     def _filter_partitions(self, topic_partitions):
         topic_partitions.sort()
 
-        return [
-            topic_partitions[num::self.num_processes]
-            for num in range(self.num_processes)
-        ][self.process_num]
+        if not self.dedicated_migration_process:
+            return [
+                topic_partitions[num::self.num_processes]
+                for num in range(self.num_processes)
+            ][self.process_num]
+        else:
+            if self.process_num == 0:
+                return None
+            else:
+                return [
+                    topic_partitions[num::self.num_processes]
+                    for num in range(self.num_processes - 1)
+                ][self.process_num - 1]
```

Comment:
This won't return the correct breakdown since we're still breaking up the `topic_partitions` list into batches of `self.num_processes`. It should be:

```
                num_processes = self.num_processes - 1
                return [
                    topic_partitions[num::num_processes]
                    for num in range(num_processes)
                ][num_processes]
```

This code is ripe some unit tests.

Reply:
thank you @snopoke, I made the changes and deploy them in staging now.